integration_tests/src/main/python/prune_partition_column_test.py

```diff
@@ -54,16 +58,22 @@ def test_prune_partition_column_when_project(spark_tmp_path, prune_part_enabled,
         conf=all_confs)
 
 
+@allow_non_gpu('ProjectExec', 'FilterExec')
 @pytest.mark.parametrize('prune_part_enabled', [False, True])
 @pytest.mark.parametrize('file_format', file_formats)
 @pytest.mark.parametrize('filter_col', ['a', 'b', 'c'])
-def test_prune_partition_column_when_project_filter(spark_tmp_path, prune_part_enabled, filter_col, file_format):
+@pytest.mark.parametrize('gpu_project_enabled', [True, False])
+@pytest.mark.parametrize('gpu_filter_enabled', [True, False])
```

Comment:
We should not be allowing ProjectExec or FilterExec to be on the CPU when the corresponding enable is False.  If it were to happen for some reason, then we're not testing what we think we're testing and the test is invalid but silently succeeds.

My original comment wasn't to literally combine the tests into a single test, rather to take the body of the tests and factor them out into reusable chunks, so then we can still annotate properly but not replicate code within each test.  For example:
```python
def test_prune_partition_column_when_project(...):
  do_prune_partition_column_test_project(...)

@allow_non_gpu("ProjectExec")
[...]
def test_prune_partition_column_when_fallback_project(..):
  do_prune_partition_column_test_project()

def test_prune_partition_column_when_filter(...):
  do_prune_parittion_column_test_filter()

@allow_non_gpu("FilterExec")
[...]
def test_prune_partition_column_when_fallback_filter(...):
  do_prune_partition_column_test_filter(project_disable_conf)

@allow_non_gpu("ProjectExec", "FilterExec")
[...]
def test_prune_partition_column_when_fallback_project_filter(...):
  do_prune_partition_column_test_filter(project_filter_disable_conf)
```

Reply:
done